### PR TITLE
Exclude virtual platform from resolution results

### PR DESCRIPTION
**Pull request by ljacomet**
When using a virtual platform, it cannot appear in the
`ResolutionResult` so it is filtered when the resulting graph is saved.

However this does not resolve the publication issue.

### CODE REVIEW

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/EdgeState.java [hunk 1]
```diff
@@ -238,6 +238,12 @@ public Long getSelected() {
         return getSelectedComponent().getResultId();
     }
 
+    @Override
+    public boolean isSelectedAVirtualPlatform() {
```
Comment (melix) [inline subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/EdgeState.java]: :o: strange method name :p Maybe `isSelectedAndVirtualPlatform()` ?

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/NodeState.java [hunk 2]
```diff
@@ -375,6 +375,10 @@ public boolean isSelected() {
         return !incomingEdges.isEmpty();
     }
 
+    public boolean isIncludedInGraphResult() {
```
Comment (melix) [inline subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/NodeState.java]: :o: I would rename to `shouldIncludeInGraphResult()`

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/EdgeState.java (hunk 1 above)
Comment (ljacomet) [reply subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/EdgeState.java]: Not exactly what I meant. What about `isTargetAVirtualPlatform()`?

Comment (melix) [reply subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/EdgeState.java]: It's the "A" that puzzles me :)

Comment (ljacomet) [reply subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/EdgeState.java]: Ok, will settle for `isTargetVirtualPlatform()` then 😉

**Pull request merged**